crates/iron-remote-desktop/src/clipboard.rs

```diff
@@ -0,0 +1,18 @@
+use wasm_bindgen::JsValue;
+use web_sys::js_sys;
+
+pub trait ClipboardTransaction {
+    type ClipboardContent: ClipboardContent;
+
+    fn init() -> Self;
+    fn add_content(&mut self, content: Self::ClipboardContent);
+    fn is_empty(&self) -> bool;
+    fn js_contents(&self) -> js_sys::Array;
+}
+
+pub trait ClipboardContent {
+    fn new_text(mime_type: &str, text: &str) -> Self;
+    fn new_binary(mime_type: &str, binary: &[u8]) -> Self;
+    fn js_mime_type(&self) -> String;
+    fn js_value(&self) -> JsValue;
```

Comment:
suggestion: The `js_` prefix was a hack because we already had a function named `contents`. Now that it’s a trait, it’s okay to reuse the "proper" name.

suggestion: Use `impl Into<JsValue>` and `&str` as return types.

```suggestion
    fn contents(&self) -> impl Into<JsValue>;
}

pub trait ClipboardContent {
    fn new_text(mime_type: &str, text: &str) -> Self;
    fn new_binary(mime_type: &str, binary: &[u8]) -> Self;
    fn mime_type(&self) -> &str;
    fn value(&self) -> impl Into<JsValue>;
```

In the exported functions, it’s then possible to do:
```rust
clipboard_content.mime_type().to_owned()
```
```rust
clipboard_content.value().into()
```

Reply:
Had the issue with `impl Into<JsValue>;`:
```bash
error[E0308]: `match` arms have incompatible types                                                                                                                                                                                 
  --> crates\ironrdp-web\src\clipboard\transaction.rs:63:54
   |
61 | /         match self {
62 | |             ClipboardContentValue::Text(text) => text.as_str(),
   | |                                                  ------------- this is found to be of type `&str`
63 | |             ClipboardContentValue::Binary(binary) => js_sys::Uint8Array::from(binary.as_slice()),
   | |                                                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `&str`, found `Uint8Array`
64 | |         }
   | |_________- `match` arms have incompatible types
   |
help: you could change the return type to be a boxed trait object
   |
60 |     pub(crate) fn value(&self) -> Box<dyn Into<JsValue>> {
   |                                   ~~~~~~~              +
help: if you change the return type to expect trait objects, box the returned expressions
   |
62 ~             ClipboardContentValue::Text(text) => Box::new(text.as_str()),
63 ~             ClipboardContentValue::Binary(binary) => Box::new(js_sys::Uint8Array::from(binary.as_slice())),
   |

For more information about this error, try `rustc --explain E0308`.
error: could not compile `ironrdp-web` (lib test) due to 1 previous error
```
But I added - `fn mime_type(&self) -> &str;`